### PR TITLE
feat(netpol tests):  netpol probeConnectivity refactor

### PR DESCRIPTION
**Pull request by crisboarna**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
 - Clean up netpol `probeConnectivity` function params

#### Special notes for your reviewer:

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
```

### CODE REVIEW

**Comment by aojea** [comment]
> * Clean up straggler/duplicate `NodeOSDistroIs` fun outside of framework.

I think is better to not mix different topics like this, better to use a different commit or PR,

**Comment by aojea** [comment]
> > * Clean up straggler/duplicate `NodeOSDistroIs` fun outside of framework.
> 
> I think is better to not mix different topics like this, better to use a different commit or PR,

Different prs better , each change address different reviewers and , if I'm not wrong, both changes are unrelated

**Comment by crisboarna** [comment]
Ok, removed `NodeOSDistroIs` and leaving this PR only for the `probeConnectivity` change.

Thank you for the guidance @aojea 

/retest

**Comment by aojea** [comment]
> Ok, removed `NodeOSDistroIs` and leaving this PR only for the `probeConnectivity` change.
> 
> Thank you for the guidance @aojea

no problem, sometimes we can conflate changes, but in this case there is no relation and I think is better to treat them separately.

Just squash both commits and force push so we only have the change to the netpol bits in the PR

**Comment by aojea** [comment]
> Use `git rebase` to reapply your commits on top of the target branch. Detailed instructions for doing so can be found [here](https://git.k8s.io/community/contributors/guide/github-workflow.md#4-keep-your-branch-in-sync).

use `git rebase -i HEAD~2` and squash, the `git push -f origin "yourbranch"`

File: test/e2e/network/netpol/probe.go [hunk 1]
```diff
@@ -108,7 +108,15 @@ func probeWorker(prober Prober, jobs <-chan *ProbeJob, results chan<- *ProbeJobR
 		// dnsName := job.PodTo.QualifiedServiceAddress(job.ToPodDNSDomain)
 
 		// TODO make this work on dual-stack clusters...
-		connected, command, err := prober.probeConnectivity(podFrom.Namespace, podFrom.Name, podFrom.Containers[0].Name(), job.PodTo.ServiceIP, job.Protocol, job.ToPort, timeoutSeconds)
+		connected, command, err := prober.probeConnectivity(&probeConnectivityArgs{
+			nsFrom:         podFrom.Namespace,
+			podFrom:        podFrom.Name,
+			containerFrom:  podFrom.Containers[0].Name(),
+			addrTo:         job.PodTo.QualifiedServiceAddress(job.ToPodDNSDomain),
```
Comment (aojea) [inline test/e2e/network/netpol/probe.go:115]: why do you change `job.PodTo.ServiceIP` to `job.PodTo.QualifiedServiceAddress(job.ToPodDNSDomain)` ?

refactors should not change current status, just reorganize the code, 
changes should be done in an independent commit so the reviewer can understand it , otherwise these things can be missed in reviews

Comment (jayunit100) [reply test/e2e/network/netpol/probe.go]: thanks for this catch @aojea  :) didnt see this PR until just now, but looks like its fixed

**Comment by jayunit100** [comment]
thanks for the cleanup @crisboarna ! i saw this late but if you want to reach out to me on slack i.e. in the sig-network room, we can talk about other things we want to do here, esp around tests for windows udp, dualstack tests, and so on -
